api/v1/net_services_utils.go

```diff
@@ -0,0 +1,172 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/celestiaorg/bittwister/xdp"
+	"github.com/celestiaorg/bittwister/xdp/bandwidth"
+	"github.com/celestiaorg/bittwister/xdp/latency"
+	"github.com/celestiaorg/bittwister/xdp/packetloss"
+	"go.uber.org/zap"
+)
+
+const ServiceStopTimeout = 5 // Seconds
+
+type netRestrictService struct {
+	service xdp.XdpLoader
+	ctx     context.Context
+	cancel  context.CancelFunc
+	logger  *zap.Logger
+}
+
+func (n *netRestrictService) Start(networkInterfaceName string) error {
+	if n.service == nil {
+		return ErrServiceNotInitialized
+	}
+
+	iface, err := net.InterfaceByName(networkInterfaceName)
+	if err != nil {
+		return fmt.Errorf("lookup network device %q: %v", networkInterfaceName, err)
```

Comment:
When using a non existing interface name. The API server logs an error, but only returns a `500` code as a response. For debugging on the user side it would be great to pass the content of the error to the requested user.